### PR TITLE
PR: Simplify custom layout definition

### LINKED ISSUE #8613: Use new functionality in Qt 5.6 to resize dockwidgets programatically
This way we could avoid the kind of hacky methods we have right now to create the Matlab, RStudio, etc, layouts.

This new change is mentioned here:

https://woboq.com/blog/qdockwidget-changes-in-56.html

### PR DESCRIPTION
**Pull request by goanpeca**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)
* [ ] Included a screenshot (if affecting the UI)

The code for setting the default layouts has been simplified to make use of Qt 5.6 `resizeDocks`

### Issue(s) Resolved

Fixes #8613

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @goanpeca

### CODE REVIEW

**Comment by goanpeca** [comment]
@CAM-Gerlach @ccordoba12 could you test this on win/linux?

Thanks

**Comment by goanpeca** [comment]
I am seeing these `QMainWidget::resizeDocks: all sizes need to be larger than 0` on the console, but no idea why, since all values are indeed larger that 0 🤷‍♂️

**Comment by CAM-Gerlach** [comment]
Sure @goanpeca , in work now.

File: spyder/app/mainwindow.py [hunk 1]
```diff
@@ -1677,25 +1693,38 @@ def setup_default_layouts(self, index, settings):
...
+        # Set widths and heights
+        width_fractions = layout['width fraction']
+        if len(width_fractions) > 1:
+            # TODO: get the first col[i] that is not None
+            _widgets = [col[0][0].dockwidget for col in widgets_layout]
+            print(_widgets, width_fractions, Qt.Horizontal)
+            self.resizeDocks(_widgets, width_fractions, Qt.Horizontal)
+
+        height_fractions = layout['height fraction']
+        for i, column in enumerate(widgets_layout_clean):
```
Comment (CAM-Gerlach) [inline spyder/app/mainwindow.py]: ``idx`` seems to be the modern convention

File: spyder/app/mainwindow.py [hunk 2]
```diff
@@ -1677,25 +1693,38 @@ def setup_default_layouts(self, index, settings):
...
+            if len(column) > 1:
+                _widgets = [row[0].dockwidget for row in column]
+                print(_widgets, height_fractions[i])
```
Comment (CAM-Gerlach) [inline spyder/app/mainwindow.py]: Is this still needed? It will fail the tests, ofc.

Comment (goanpeca) [reply spyder/app/mainwindow.py]: Oops

File: spyder/app/mainwindow.py (hunk 1 above)
Comment (goanpeca) [reply spyder/app/mainwindow.py]: Agreed ;-)

**Comment by goanpeca** [comment]
Thanks for the review @CAM-Gerlach, I will need to check on my Linux VM.

I have tested on Mac only so far.

**Comment by CAM-Gerlach** [comment]
Also, @goanpeca , you should affirm the DCO at the bottom of the PR template.

**Comment by goanpeca** [comment]
Need to install a VM to check tis on linux :-|

**Comment by goanpeca** [comment]
Since I had to delete the fork, I have recreated this PR

**Comment by CAM-Gerlach** [comment]
> Since I had to delete the fork, I have recreated this PR

Github won't pick it up again if you simply push to the same fork name?

**Comment by goanpeca** [comment]
Nope it seems

Screen Shot 2019-04-14 at 11 16 31

**Comment by CAM-Gerlach** [comment]
Gotcha, thanks @goanpeca . I'd been wondering that.

**Pull request closed without merging**